functions/Get-DbaInstallDate.ps1

```diff
@@ -0,0 +1,188 @@
+
+Returns an object with SQL Server Install date as a string and the Windows install date as string for every server listed in the Central Management Server on sql2014
+	
+#>
+	[CmdletBinding(DefaultParameterSetName = "Default")]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance", "ComputerName")]
+		[object[]]$SqlServer,
+		[parameter(ParameterSetName = "Sql")]
+		[Switch]$SqlOnly,
+		[parameter(ParameterSetName = "Windows")]
+		[Switch]$WindowsOnly,
+		[Alias("Credential")]
+		[PsCredential]$SqlCredential,
+		[PsCredential]$WindowsCredential
+	)
+	
+	PROCESS
+	{
+		foreach ($instance in $SqlServer)
+		{
+			if ($instance.Gettype().FullName -eq [System.Management.Automation.PSCustomObject] )
+			{
+				$servername = $instance.SqlInstance
+			}
+			elseif ($instance.Gettype().FullName -eq [Microsoft.SqlServer.Management.Smo.Server])
+			{
+				$servername = $instance.NetName
+			}
+			else
+			{
+				$servername = $instance
+			}
+						
+			if ($WindowsOnly -ne $true)
+			{ 
+				try 
+				{
+					Write-Message -Level Verbose -Message "Connecting to $instance" -Silent $false 
+                    if ($SqlCredential)
+                    { 
+                        $server = Connect-SqlInstance -SqlInstance $servername -SqlCredential $SqlCredential #-ErrorVariable ConnectError
+                    } else 
+                    { 
+                        $server = Connect-SqlInstance -SqlInstance $servername 
+                    }					
+				}
+				catch 
+				{
+					Stop-Function -Message "Failed to connect to: $instance" -Continue -Target $instance					
+				}
+
+				if ( $server.VersionMajor -ge 9 )
+				{ 
+					Write-Message -Level Verbose -Message "Getting Install Date for: $instance" -Silent $false 
+					$sql = "SELECT create_date FROM sys.server_principals WHERE sid = 0x010100000000000512000000"
+					$sqlInstallDate = $server.databases['master'].ExecuteWithResults($sql).tables
+					$sqlInstallDate = ($sqlInstallDate.rows).create_date.toString('MM-dd-yyyy') 
+
+				} else { 
+					Write-Message -Level Verbose -Message "Getting Install Date for: $instance" -Silent $false 
+					$sql = "SELECT schemadate FROM sysservers"
+					$sqlInstallDate = $server.databases['master'].ExecuteWithResults($sql).tables					
+					$sqlInstallDate = ($sqlInstallDate.rows).create_date.toString('MM-dd-yyyy') 
+				}											
+
+			} 
+
+			if ($SqlOnly -ne $true)			
+			{ 
+				Write-Message -Level Verbose -Message "Getting Windows Server Name for: $servername" -Silent $false 
+				$WindowsServerName = (Resolve-DbaNetworkName $servername -Credential $WindowsCredential).ComputerName
+				
+				try
+				{
+					Write-Message -Level Verbose -Message "Getting Windows Install date via CIM for: $WindowsServerName" -Silent $false 
+					$windowsInstallDate = (Get-CimInstance -ClassName win32_operatingsystem -ComputerName $windowsServerName -ErrorAction SilentlyContinue).InstallDate	
```

Comment:
There are new Computer Management functions included to handle calling CIM/WMI properly and include bunch of backend support for failures on one protocol and such.

These are in PR #1133 for reference as I am still working to figure out proper use.